pkg/pillar/zfs/zfs.go

```diff
@@ -151,10 +154,10 @@ func GetZvolPath(datasetName string) string {
 }
 
 // CreateVaultVolumeDataset Create an empty vault zvol
-func CreateVaultVolumeDataset(log *base.LogObject, datasetName string, zfsKeyFile string, encrypted bool, sizeBytes uint64) error {
-	// Shave off reserved + Can't align up if we're already at max space.
-	sizeBytes = sizeBytes - reserveEveStorageSizeGb - (volBlockSize * 1024)
-	alignedSize := alignUpToBlockSize(sizeBytes)
+func CreateVaultVolumeDataset(log *base.LogObject, datasetName string, zfsKeyFile string, encrypted bool, sizeBytes uint64, compressionType string, blockSize uint64) error {
+	// Shave off 1MB, Can't align up if we're already at max space.
```

Comment:
Is the blocksize 16kByte (the above default)? or something else?

And why do we need this? What's the relationship to the etcd vault?

Reply:
Yes 16KB is the default, only etcd requests a different block size.